cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java

```diff
@@ -129,20 +129,22 @@ protected Event doExecute(final RequestContext context) throws Exception {
             final BaseClient<Credentials, CommonProfile> client =
                     (BaseClient<Credentials, CommonProfile>) this.clients
                     .findClient(clientName);
-            logger.debug("client : {}", client);
+            logger.debug("client: {}", client);
 
-            // HTTP protocol not allowed
-            if (client.getProtocol() == Protocol.HTTP) {
-                throw new TechnicalException("HTTP protocol client not supported : " + client);
+            // Allow CAS, OAuth, OpenID and SAML protocols
+            final Mechanism mechanism = client.getMechanism();
+            if (mechanism != Mechanism.CAS_PROTOCOL && mechanism != Mechanism.OAUTH_PROTOCOL
```

Comment:
Suggestion: Create a list of all supported protocols as a const, and do `list.contains(mechanism)=false` instead.